### ISSUE TITLE
Support structured outputs

### ISSUE THREAD

**Issue by idjuricTT** [tenstorrent/tt-inference-server#2225]
Special case of sampling where server dynamically takes output logits and prevents some tokens from being returned

**Comment by rreece** [comment]
I want to highlight that this is a requirement for the BetterBrain demo

There's some discussion of there demo requirements in this doc:
https://docs.google.com/document/d/1gv8ClMmSJlFyEdQC9HpfkmfIINN3MX5ZMhVhQHwDmQQ/edit?usp=sharing

The TLDR is really that we should provide them an OpenAI-compatible LLM chat that supports **Structured Outputs**.

This enables their application to do tool-calling on their side. So we don't need tool-calling support for this demo, but we do need Structured Outputs.

**Comment by rreece** [comment]
Besides the server supporting the spec, I’m pretty sure you need guided decode kernel support in the tt-metal sampling.

**Comment by rreece** [comment]
Related:
-   https://github.com/tenstorrent/cloud/issues/5964
-   https://github.com/tenstorrent/cloud/issues/5953
